server/agents/helpers/word_frequency_percentiles.py

```diff
@@ -22,21 +22,26 @@ def load_frequency_list(language_code: str) -> pd.DataFrame:
 
     # Construct the path to the frequency list
     folder_name = language_code_map[language_code]
-    file_path = os.path.join("./word_frequency_lists/content/2018", folder_name, f"{folder_name}_50k.txt")
+    file_path = os.path.join("./word_frequency_lists/content/2018", folder_name, f"{folder_name}_{type}.txt")
 
     # Load the frequency list into a DataFrame
     df = pd.read_csv(file_path, sep=" ", names=['word', 'frequency'], header=None)
     df['rank'] = df['frequency'].rank(ascending=False)
     return df
 
 def get_word_frequency_percentiles(transcript: str, language_code: str) -> dict:
-    df = load_frequency_list(language_code)
+    df = load_frequency_list(language_code, '50k')
     words_dict = pd.Series(df['rank'].values,index=df['word']).to_dict()
     total_words = len(df)  # Get the total number of words in the dataset
 
+    df = load_frequency_list(language_code, 'full')
```

Comment:
50k will have us covered 99% of the time. Therefore, I suggest you only load up the full frequency list as a last resort, i.e. after you can't find a word.

Reply:
I agree, and more so, we should only load this all once. I added that in my commit.